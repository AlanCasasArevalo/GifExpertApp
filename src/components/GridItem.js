import React from 'react'
import PropTypes from 'prop-types'

// const GridItem = ({category}) => {
const GridItem = ({id, title, url }) => {

    // const [images, setImages] = useState([])

    return (
        <div className="card">
            <img src={url} alt={title}/>
            <p>{title}</p>
        </div>
    )
}
//
// GridItem.propTypes = {
//     image: PropTypes.object.isRequired,
// }

export default GridItem;
