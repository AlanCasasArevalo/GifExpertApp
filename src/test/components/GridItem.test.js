import {shallow} from "enzyme";
import React from "react";
import GridItem from "../../components/GridItem";

describe('GridItem Component test', () => {

    const title = 'Hola, soy Alan'
    const url = 'http://localhost:3000/image'
    let wrapper = shallow(
        <GridItem
            title={title}
            url={url}
        />)


    beforeEach(() => {
        wrapper = shallow(
            <GridItem
                title={title}
                url={url}
            />)
    })

    test('GridItem should be defined', () => {
        expect(<GridItem />).toBeDefined()
    })
    test('GridItem should be an object', () => {
        expect(typeof <GridItem />).toBe('object')
    })

    test('GridItem should return a snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    })
    test('GridItem should return title by props', () => {
        const titleInP = wrapper.find('p').text(title)
        expect(title).toBe(titleInP)
    })


    test('GridItem should return url and title into image by props', () => {
        const image = wrapper.find('img')
        const { src, alt } = image.props()
        expect(url).toStrictEqual(src)
        expect(title).toStrictEqual(alt)
    })

    test('GridItem should get animate__flash class into div', () => {
        const div = wrapper.find('div')
        const {className, children} = div.props()
        expect(className).toStrictEqual('card animate__animated animate__flash')
    })

    test('div should get image and p', () => {
        const div = wrapper.find('div')
        const {className, children} = div.props()
        expect(children[0].type).toStrictEqual('img')
        expect(children[1].type).toStrictEqual('p')
    })
})
