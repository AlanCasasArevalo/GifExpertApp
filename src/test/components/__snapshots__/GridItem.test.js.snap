// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridItem Component test GridItem should return a snapshot 1`] = `
<div
  className="card animate__animated animate__flash"
>
  <img
    alt="Hola, soy Alan"
    src="http://localhost:3000/image"
  />
  <p>
    Hola, soy Alan
  </p>
</div>
`;
